### ISSUE TITLE
deadlock in deadlock detector

### ISSUE THREAD

**Issue by zzlk** [hydro-project/hydroflow#810]
~~I am not sure but I think the issue is related to spawning a process and then trying to read from stdout too quickly can cause the read to immediately return with EOF which causes the tests to fail non-deterministically. Likely the fix is to add a small delay in the spawn_child function before returning control back to the test. But both the issue and fix here are speculated.~~

I confirmed that this is a bug with the deadlock detector example code below.

example:
https://github.com/hydro-project/hydroflow/actions/runs/5449850115/jobs/9914512538

**Comment by zzlk** [comment]
This is actually a deadlock inside the deadlock detector, it gets stuck spinning in the hydroflow graph.

**Comment by zzlk** [comment]
Here is a mermaid graph showing what it should look like: [mermaid](https://mermaid.live/edit#pako:eNqVWF1zmzgU_SsqnTZ4BrtGAmOTj53d7Wt3OpNOH7buMAoIhw0gF0TSNPF_X0lIAn_U2HnIoGvpXOnq3HsuvFgxTYgVWu_evWRlxsKXC3ZPCnIRXtzhmlw47fArrjJ8l5P6gk-I86ZmpPrrYcVnvU2ShM_SNlolpBLm-Xx-sdls3r1blmlOn-J7XDHw5eOyjHNc1x9JCtZNnv8tBiDN8jx8O8epU7OKPpDw7XQ6dRj5ycY4z1ZlmJOUOU_3GSPjeo1jEq4rsoVU3_eR0nR-HlKelQ-37DknICEpbnIG9HKMsYIaP2UJuw-99U8npjmtwookTkpLNq6zXyR0Jz4pLpdl3dytKry-B_Uqch9d8G1pqScOg1lTAHdpfV-WgP-5i0f323K5tK6S7JH_zjdw3dtrTEoe0htbTBtdfeBzbsCVuK6blOZJGL7P2eXVXXUjoC5qhlkWO2L8fsUubf3DJ1LXeEXCsCRP9sjR5tciBO9xsb4sGqbnOIAkK_IKXvQk8VdMhLGeZGVNKmaLwch43Ug8vje5q6X14XsYhuZe1SGn6pC2eBqpExR4bWcJP2DGno-vn-vl8251XNG6jv6jWWkfX6wDbC-6xZwjPKz2a_QKVqSM7ijN7elkPjqO5Gkkb_sQr3bhADx6BTaPT8Yv7heJinplFyNhPo7pc8wPHNLvIBNSs6jmdLRpw-5oUyYdxHLZYiiydxwaj8c3IrhdwMfjqTDOtW3eTlrosVrk6bHXjn093mFx9IiraMVjnq0FofkIXDF201quPvBHw2m9g3ZEymQIUF7jPqo070ObE6lTnOBGBzLiNWjL0dYP-668vit_y5X81_cFTabDXqZPu0w3OdBddU2bKiZRJsi4JqSK8qxmxwkDNQzcIWH9CurJGlc1acvCLY0fCPszSSpZDEaTpnzie7UHCIkUIVEHzwixhyjoqmVu73QprSKC43u5t3WVlSwv39jvfzSUXX7mpw3BS_jHph07oB4N-WgJCvU1wHaM9BiJMZSJ4B4iMVRkEIGu-ySQhgMU7l8-7A_QcSYgwwR0mAnmDl14mAvf7NH3gZKq78lFhwIe7Uf8y_OaAJ5mPAFEBQe4BhiwZs0Vj6aAPVH-IyMrERv7p_M84uIg1w1ei7oHV0ZlLxieCYZ3MBgzHYvZXij4ZIILOyt3iuCheAQaJthOC15M-zW5TQ0ldiIvBqJsSr7r_W57NUuykiduSeoBMN-A-btSFMkUFiBce4HM4kjcki1MXeaCzUBxMEILe0Kb5pi1DopXLeYDWzW34vaupSkzOqS2bqB5GZxVQGYtiQKdWYEYezLfO1VTEuWaSiyelM6JjXZB0IXAGF3lwA0OlQZPlQYZmn5pkIYDpWHWLwcd5kER0uCKx3sa1Lfvu9rydJoj3ubtn8QYD5xmS-bcAZ3zTUL7h6ub6dbcXrtW8p424t3pwxH-7LmaGVezg65gJ4a7aminFeVdGaOiLdPPDrjNCl7wPmPexctu-JHEb77pn7-PBrQRGg2H7nlpAWcntfli2k6b3ysPvXOs-QlGvTY9S6VpEj_HOc81XkP-oSUBG0DymvDRLS2IbYuVCkQCiIIiEIaqiimuMDglzENRRAaup10ndPLQOy2I3m4Q261GRZbw86ttipceGYR2x71YykCK8sTj1b3n7Ab_lLjp5h76Z9VCOG_XDRxzfg5XQmA3CDpA_uuyQKghN0kl3KaTWA-urzmCpk9LMFzSMotxLtVI06sl205IDp9toWOy2MqgH81gWND0lLCIWbtvyrorEicIwVcS947dv_jthknMBqkozbxRNW0qRyu4nLaJ1mbR5SkHRzKJeF_dSyFe8BOapmahWEbZPal21o3HbedrOlIIje71jLpB7pRw1k0zPSvsAXZGpNYaKYBKbKGRA9iJLZxvG9tG3KgTVO-a0LydQfW2iaaH5HfWEy4R0j3hksZ94er2r3fcG_lHtVK73HP3O1fucVUMjFQFPamCBgMZVUQ9VdSfP4ZEEXVBlvtwA3MVCHWGdobXGWQThPzOgKRBEsRVl4Tkqw1yFT3d0-mJXONU8gFBhQHPwIAGo22qEFIg6AwQxV7UbsNTCN4ZCHzu9fU1dSVjka8Q_DMQfMX5NpwzhTA7A0G1qFDmEdLX0-5oofAWZ-Dpz0IcyXKsgvDClSVWaL0ISi4t-VV3aYX8UX3y5KWr3PCpuGH09rmMrZBVDXGsZp1gRj5mmNO9sMIU86rvWGtc_ktpoSfxoRW-WD-tcBws4MSfzqZzL_CnrhtAx3oWZn82QVN3EcyD2Szwpt7GsX5JBMjt_mK28BdTuFgspgFyLJJkjFaf2k_T8gv15n8F-wf9)

here is a sample from the scheduler tracer showing it get stuck:

```
2023-07-14T00:13:25.692278Z TRACE run_async:run_available_async:run_stratum{tick=839 stratum=1}: hydroflow::scheduled::graph: Running subgraph. sg_id=3
2023-07-14T00:13:25.692327Z TRACE run_async:run_available_async:run_stratum{tick=839 stratum=1}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.692373Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting `next_stratum` call. events_received_tick=true can_start_tick=false
2023-07-14T00:13:25.692389Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=839 stratum=1
2023-07-14T00:13:25.692403Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=839 stratum=2
2023-07-14T00:13:25.692417Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Work found on stratum. tick=839 stratum=2
2023-07-14T00:13:25.692430Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.749048Z TRACE run_async:run_available_async:run_stratum{tick=839 stratum=2}: hydroflow::scheduled::graph: Running subgraph. sg_id=6
2023-07-14T00:13:25.749111Z TRACE run_async:run_available_async:run_stratum{tick=839 stratum=2}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.749200Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting `next_stratum` call. events_received_tick=true can_start_tick=true
2023-07-14T00:13:25.749217Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=839 stratum=2
2023-07-14T00:13:25.749232Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: End of tick 839, starting tick 840. can_start_tick=true
2023-07-14T00:13:25.749256Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}:try_recv_events{events_received_tick=false}: hydroflow::scheduled::graph: Event received. sg_id=3 is_external=false sg_stratum=1
2023-07-14T00:13:25.749275Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}:try_recv_events{events_received_tick=false}: hydroflow::scheduled::graph: return=1
2023-07-14T00:13:25.749298Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: `can_start_tick` is `true`, continuing. tick=840
2023-07-14T00:13:25.749311Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=840 stratum=0
2023-07-14T00:13:25.749326Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Work found on stratum. tick=840 stratum=0
2023-07-14T00:13:25.749340Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.749366Z TRACE run_async:run_available_async:run_stratum{tick=840 stratum=0}: hydroflow::scheduled::graph: Running subgraph. sg_id=4
2023-07-14T00:13:25.806648Z TRACE run_async:run_available_async:run_stratum{tick=840 stratum=0}: hydroflow::scheduled::graph: Running subgraph. sg_id=2
2023-07-14T00:13:25.806767Z TRACE run_async:run_available_async:run_stratum{tick=840 stratum=0}: hydroflow::scheduled::graph: Running subgraph. sg_id=5
2023-07-14T00:13:25.806835Z TRACE run_async:run_available_async:run_stratum{tick=840 stratum=0}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.806946Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting `next_stratum` call. events_received_tick=true can_start_tick=false
2023-07-14T00:13:25.806962Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting tick, setting `can_start_tick = false`.
2023-07-14T00:13:25.806973Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=840 stratum=0
2023-07-14T00:13:25.806986Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=840 stratum=1
2023-07-14T00:13:25.806999Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Work found on stratum. tick=840 stratum=1
2023-07-14T00:13:25.807012Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.807039Z TRACE run_async:run_available_async:run_stratum{tick=840 stratum=1}: hydroflow::scheduled::graph: Running subgraph. sg_id=3
2023-07-14T00:13:25.807070Z TRACE run_async:run_available_async:run_stratum{tick=840 stratum=1}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.807112Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting `next_stratum` call. events_received_tick=true can_start_tick=false
2023-07-14T00:13:25.807127Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=840 stratum=1
2023-07-14T00:13:25.853760Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=840 stratum=2
2023-07-14T00:13:25.853831Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Work found on stratum. tick=840 stratum=2
2023-07-14T00:13:25.853847Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.853919Z TRACE run_async:run_available_async:run_stratum{tick=840 stratum=2}: hydroflow::scheduled::graph: Running subgraph. sg_id=6
2023-07-14T00:13:25.853946Z TRACE run_async:run_available_async:run_stratum{tick=840 stratum=2}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.854039Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting `next_stratum` call. events_received_tick=true can_start_tick=true
2023-07-14T00:13:25.854056Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=840 stratum=2
2023-07-14T00:13:25.854071Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: End of tick 840, starting tick 841. can_start_tick=true
2023-07-14T00:13:25.854095Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}:try_recv_events{events_received_tick=false}: hydroflow::scheduled::graph: Event received. sg_id=3 is_external=false sg_stratum=1
2023-07-14T00:13:25.854115Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}:try_recv_events{events_received_tick=false}: hydroflow::scheduled::graph: return=1
2023-07-14T00:13:25.854138Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: `can_start_tick` is `true`, continuing. tick=841
2023-07-14T00:13:25.854151Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=841 stratum=0
2023-07-14T00:13:25.914256Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Work found on stratum. tick=841 stratum=0
2023-07-14T00:13:25.914331Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.914401Z TRACE run_async:run_available_async:run_stratum{tick=841 stratum=0}: hydroflow::scheduled::graph: Running subgraph. sg_id=4
2023-07-14T00:13:25.914426Z TRACE run_async:run_available_async:run_stratum{tick=841 stratum=0}: hydroflow::scheduled::graph: Running subgraph. sg_id=2
2023-07-14T00:13:25.914475Z TRACE run_async:run_available_async:run_stratum{tick=841 stratum=0}: hydroflow::scheduled::graph: Running subgraph. sg_id=5
2023-07-14T00:13:25.914560Z TRACE run_async:run_available_async:run_stratum{tick=841 stratum=0}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.914654Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting `next_stratum` call. events_received_tick=true can_start_tick=false
2023-07-14T00:13:25.914671Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting tick, setting `can_start_tick = false`.
2023-07-14T00:13:25.914683Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=841 stratum=0
2023-07-14T00:13:25.914697Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=841 stratum=1
2023-07-14T00:13:25.914711Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Work found on stratum. tick=841 stratum=1
2023-07-14T00:13:25.914725Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.914753Z TRACE run_async:run_available_async:run_stratum{tick=841 stratum=1}: hydroflow::scheduled::graph: Running subgraph. sg_id=3
2023-07-14T00:13:25.976110Z TRACE run_async:run_available_async:run_stratum{tick=841 stratum=1}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.976269Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting `next_stratum` call. events_received_tick=true can_start_tick=false
2023-07-14T00:13:25.976287Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=841 stratum=1
2023-07-14T00:13:25.976299Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=841 stratum=2
2023-07-14T00:13:25.976311Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Work found on stratum. tick=841 stratum=2
2023-07-14T00:13:25.976323Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.976349Z TRACE run_async:run_available_async:run_stratum{tick=841 stratum=2}: hydroflow::scheduled::graph: Running subgraph. sg_id=6
2023-07-14T00:13:25.976365Z TRACE run_async:run_available_async:run_stratum{tick=841 stratum=2}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:25.976402Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting `next_stratum` call. events_received_tick=true can_start_tick=true
2023-07-14T00:13:25.976417Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=841 stratum=2
2023-07-14T00:13:25.976429Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: End of tick 841, starting tick 842. can_start_tick=true
2023-07-14T00:13:25.976448Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}:try_recv_events{events_received_tick=false}: hydroflow::scheduled::graph: Event received. sg_id=3 is_external=false sg_stratum=1
2023-07-14T00:13:26.062095Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}:try_recv_events{events_received_tick=false}: hydroflow::scheduled::graph: return=1
2023-07-14T00:13:26.062172Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: `can_start_tick` is `true`, continuing. tick=842
2023-07-14T00:13:26.062184Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=842 stratum=0
2023-07-14T00:13:26.062195Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Work found on stratum. tick=842 stratum=0
2023-07-14T00:13:26.062206Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:26.062242Z TRACE run_async:run_available_async:run_stratum{tick=842 stratum=0}: hydroflow::scheduled::graph: Running subgraph. sg_id=4
2023-07-14T00:13:26.062260Z TRACE run_async:run_available_async:run_stratum{tick=842 stratum=0}: hydroflow::scheduled::graph: Running subgraph. sg_id=2
2023-07-14T00:13:26.062297Z TRACE run_async:run_available_async:run_stratum{tick=842 stratum=0}: hydroflow::scheduled::graph: Running subgraph. sg_id=5
2023-07-14T00:13:26.062348Z TRACE run_async:run_available_async:run_stratum{tick=842 stratum=0}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:26.062415Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting `next_stratum` call. events_received_tick=true can_start_tick=false
2023-07-14T00:13:26.062427Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting tick, setting `can_start_tick = false`.
2023-07-14T00:13:26.062435Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=842 stratum=0
2023-07-14T00:13:26.124725Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=842 stratum=1
2023-07-14T00:13:26.124802Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Work found on stratum. tick=842 stratum=1
2023-07-14T00:13:26.124813Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:26.124878Z TRACE run_async:run_available_async:run_stratum{tick=842 stratum=1}: hydroflow::scheduled::graph: Running subgraph. sg_id=3
2023-07-14T00:13:26.124928Z TRACE run_async:run_available_async:run_stratum{tick=842 stratum=1}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:26.125017Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting `next_stratum` call. events_received_tick=true can_start_tick=false
2023-07-14T00:13:26.125029Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=842 stratum=1
2023-07-14T00:13:26.125040Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=842 stratum=2
2023-07-14T00:13:26.125050Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Work found on stratum. tick=842 stratum=2
2023-07-14T00:13:26.125060Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:26.125081Z TRACE run_async:run_available_async:run_stratum{tick=842 stratum=2}: hydroflow::scheduled::graph: Running subgraph. sg_id=6
2023-07-14T00:13:26.125095Z TRACE run_async:run_available_async:run_stratum{tick=842 stratum=2}: hydroflow::scheduled::graph: return=true
2023-07-14T00:13:26.125126Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Starting `next_stratum` call. events_received_tick=true can_start_tick=true
2023-07-14T00:13:26.185102Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: Looking for work on stratum. tick=842 stratum=2
2023-07-14T00:13:26.185173Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: End of tick 842, starting tick 843. can_start_tick=true
2023-07-14T00:13:26.185217Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}:try_recv_events{events_received_tick=false}: hydroflow::scheduled::graph: Event received. sg_id=3 is_external=false sg_stratum=1
2023-07-14T00:13:26.185231Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}:try_recv_events{events_received_tick=false}: hydroflow::scheduled::graph: return=1
2023-07-14T00:13:26.185260Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::scheduled::graph: `can_start_tick` is `true`, continuing. tick=843
2023-07-14T00:13:26.185270Z TRACE run_async:run_available_async:next_stratum{current_tick_only=false}: hydroflow::sche
```

**Comment by jhellerstein** [comment]
closed in #855
